### PR TITLE
Update "MagicNumber" to not mark "Named Arguments" as code smells

### PR DESCRIPTION
**Pull request by DmytroTroynikov**

### CODE REVIEW

File: detekt-test/src/main/kotlin/io/gitlab/arturbosch/detekt/test/CompileExtensions.kt [hunk 1]
```diff
@@ -1,11 +1,12 @@
 package io.gitlab.arturbosch.detekt.test
 
+import org.intellij.lang.annotations.Language
 import java.nio.file.Path
 
 /**
  * Use this method if you define a kt file/class as a plain string in your test.
  */
-fun compileContentForTest(content: String) = KtTestCompiler.compileFromContent(content)
+fun compileContentForTest(@Language("kotlin") content: String) = KtTestCompiler.compileFromContent(content)
```
Comment (schalkms) [inline detekt-test/src/main/kotlin/io/gitlab/arturbosch/detekt/test/CompileExtensions.kt]: Why is this annotation needed?

File: detekt-rules/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/style/MagicNumberSpec.kt [hunk 2]
```diff
@@ -381,4 +381,64 @@ class MagicNumberSpec : Spek({
 			assertThat(MagicNumber().lint(code)).isEmpty()
 		}
 	}
+
+	given("ignoring named arguments") {
+		given("in constructor invocation") {
+			fun code(integer: String) = compileContentForTest("""
```
Comment (schalkms) [inline detekt-rules/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/style/MagicNumberSpec.kt]: `integer: String` is a bit misleading ;)

File: detekt-test/src/main/kotlin/io/gitlab/arturbosch/detekt/test/CompileExtensions.kt (hunk 1 above)
Comment (DmytroTroynikov) [reply detekt-test/src/main/kotlin/io/gitlab/arturbosch/detekt/test/CompileExtensions.kt]: It's a hint for Idea to interpret the strings passed into the parameter as strings of Kotlin language
image
As you can see it highlights the string accordingly

File: detekt-rules/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/style/MagicNumberSpec.kt (hunk 2 above)
Comment (DmytroTroynikov) [reply detekt-rules/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/style/MagicNumberSpec.kt]: Will update, thanks

File: detekt-test/src/main/kotlin/io/gitlab/arturbosch/detekt/test/CompileExtensions.kt (hunk 1 above)
Comment (arturbosch) [reply detekt-test/src/main/kotlin/io/gitlab/arturbosch/detekt/test/CompileExtensions.kt]: Thats interesting and a cool feature I think :)

Comment (schalkms) [reply detekt-test/src/main/kotlin/io/gitlab/arturbosch/detekt/test/CompileExtensions.kt]: I agree. Maybe a comment is appropriate here.

**Comment by Mauin** [comment]
Resolves #415

File: detekt-rules/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/style/MagicNumber.kt [hunk 3]
```diff
@@ -36,6 +38,7 @@ class MagicNumber(config: Config = Config.empty) : Rule(config) {
 	private val ignoreHashCodeFunction = valueOrDefault(IGNORE_HASH_CODE, false)
 	private val ignoreAnnotation = valueOrDefault(IGNORE_ANNOTATION, false)
 	private val ignorePropertyDeclaration = valueOrDefault(IGNORE_PROPERTY_DECLARATION, false)
+	private val ignoreNamedArgument = valueOrDefault(IGNORE_NAMED_ARGUMENT, true)
```
Comment (Mauin) [inline detekt-rules/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/style/MagicNumber.kt]: I think this default should still be `false`. If the config does not contain this argument we shouldn't assume it to be true.

File: detekt-rules/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/style/MagicNumberSpec.kt (hunk 2 above)
Comment (arturbosch) [reply detekt-rules/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/style/MagicNumberSpec.kt]: I removed the annotation. Intellij maked us fix inspections in the test code.

File: detekt-rules/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/style/MagicNumber.kt (hunk 3 above)
Comment (arturbosch) [reply detekt-rules/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/style/MagicNumber.kt]: Agree, turned it off.

File: detekt-test/src/main/kotlin/io/gitlab/arturbosch/detekt/test/CompileExtensions.kt (hunk 1 above)
Comment (arturbosch) [reply detekt-test/src/main/kotlin/io/gitlab/arturbosch/detekt/test/CompileExtensions.kt]: I removed the annotation. Intellij maked us fix inspections in the test code.

**Comment by arturbosch** [comment]
I manually merged and fixed the issues. Sorry for the inconvenience, merge conflicts started to begin.

**Pull request merged**